### PR TITLE
[feat] add plan option to browse command

### LINKED ISSUE #343: Add tdp `browse --plan` option
Would be nice to add a `--plan` option to the `browse` command to easily display the planned deployment if exist.

### PR DESCRIPTION
**Pull request by jpsavy**
#### Which issue(s) this PR fixes

Fixes #343 

#### Additional comments

Add optional type to Deployment Log type in the get_planned_deployment_log return

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: tdp/cli/commands/browse.py [hunk 1]
```diff
@@ -19,6 +19,12 @@
 @click.command(short_help="Browse deployment logs")
 @click.argument("deployment_id", required=False)
 @click.argument("operation", required=False)
+@click.option(
+    "-pl",
+    "--plan",
+    is_flag=True,
+    help="Get the PLANNED Deployment log if exist",
```
Comment (PaulFarault) [inline tdp/cli/commands/browse.py:26]:
```suggestion
    help="Print the planned deployment, if exist.",
```

File: tdp/cli/commands/browse.py [hunk 2]
```diff
@@ -35,20 +41,27 @@
...
     with session_class() as session:
         try:
-            if not deployment_id:
-                print_formatted_deployments(get_deployments(session, limit, offset))
+            if plan:
+                deployment_plan = get_planned_deployment_log(session)
+                if deployment_plan:
+                    print_formatted_deployment(deployment_plan)
+                else:
+                    print(f"Not deployment plan existing")
```
Comment (PaulFarault) [inline tdp/cli/commands/browse.py:54]:
```suggestion
                    print("No deployment plan to show.")
```

**Review by PaulFarault — CHANGES_REQUESTED** [review]
LGTM, just a few suggestions about messages.

File: tdp/cli/commands/browse.py (hunk 1 above)
Comment (jpsavy) [reply tdp/cli/commands/browse.py]: Done

**Review by PaulFarault — APPROVED** [review]

**Comment by mdrutel** [comment]
For the option, I think we should have ```-p``` instead of ```-pl```, or only ```--plan```

**Comment by sergkudinov** [comment]
Make output more pretty. Add empty lines at least...

image

File: tdp/cli/commands/browse.py [hunk 3]
```diff
@@ -35,20 +41,27 @@
 )
 @database_dsn
 def browse(
-    deployment_id: int, operation: str, limit: int, offset: int, database_dsn: str
+    deployment_id: int, operation: str, plan: bool,limit: int, offset: int, database_dsn: str
```
Comment (sergkudinov) [inline tdp/cli/commands/browse.py:44]: apply file autoformatting before committing. on my Mac it is `Shift + Option + F`.

File: tdp/cli/commands/browse.py [hunk 4]
```diff
@@ -19,6 +19,12 @@
 @click.command(short_help="Browse deployment logs")
 @click.argument("deployment_id", required=False)
 @click.argument("operation", required=False)
+@click.option(
+    "-pl",
```
Comment (sergkudinov) [inline tdp/cli/commands/browse.py:23]: change to `-p`

**Review by sergkudinov — CHANGES_REQUESTED** [review]

File: tdp/cli/commands/browse.py (hunk 3 above)
Comment (jpsavy) [reply tdp/cli/commands/browse.py]: Done.

File: tdp/cli/commands/browse.py (hunk 4 above)
Comment (jpsavy) [reply tdp/cli/commands/browse.py]: Done.

**Comment by jpsavy** [comment]
> Make please the output more pretty:
> 
> - add empty lines between blocks
> - do we need Sources and Targets printed again in separate blocks?
> - when Service Component logs is emty, it prints nothing, even without None. It is not obvious

This PR aims to add a `--plan` option to the `tdp browse` command. Hence, this comment is out of scope. You should open another issue if you want to fix it.

**Comment by sergkudinov** [comment]
In the original issue #343, there is nothing precised about the view of the output. So, this is a brain new feature and we can discuss everything in the same scope. In another hand, these are little improvements that don't worth a separate workflow.

**Review by mdrutel — APPROVED** [review]

**Comment by rpignolet** [comment]
> In the original issue #343, there is nothing precised about the view of the output. So, this is a brain new feature and we can discuss everything in the same scope. In another hand, these are little improvements that don't worth a separate workflow.

This PR aims to add the `--plan` option. What you are asking for is a change in the display of the `tdp browse` command and is therefore not within the scope of this PR. You can make an issue on this topic.

**Review by rpignolet — APPROVED** [review]

**Review by sergkudinov — APPROVED** [review]

**Pull request merged**